service_worker/background.js

```diff
@@ -1,107 +1,73 @@
-const restartTimer = async () => {
-    // clearTimeout(timerId);
-    console.log("restart timer")
-
-    await new Promise((resolve, reject) => {
-        chrome.storage.local.get(['timeLeft'], (data) => {
-            if (chrome.runtime.lastError) {
-                reject(chrome.runtime.lastError);
-            } else {
-                resolve(data);
-            }
-
-            timeoutDuration = data.timeLeft;
-            console.log(timeoutDuration, " timeoutduration")
-        });
-    });
-    startSessionTimeout()
+const startTimer = async () => {
+    try {
+        const data = await chrome.storage.local.get(['timeLeft', 'loggedIn', 'sessionTimeout'])
+        if (!data.loggedIn || data.sessionTimeout) return;
+        if (data.timeLeft) {
+            timerId = setTimeout(sessionTimeout, data.timeLeft);
+            updateTimeInLocalStorage(data.timeLeft);
+        }
+    }
+    catch (error) {
+        console.error("error starting timer ", error)
+    }
 }
 
-
-// Listener for when a window is created
-chrome.windows.onCreated.addListener(async () =>{
-    const data = await new Promise((resolve, reject) => {
-        chrome.storage.local.get(['loggedIn', 'sessionTimeout','timeLeft'], (data) => {
-            if (chrome.runtime.lastError) {
-                reject(chrome.runtime.lastError);
-            } else {
-                resolve(data);
-            }
+chrome.tabs.onCreated.addListener(async (tab) => {
+    const data = await chrome.storage.local.get(['loggedIn', 'sessionTimeout'])
+    if (data.loggedIn) {
+        chrome.action.setIcon({
+            path: "../assets/Logo_active.png",
         });
-    });
-    chrome.windows.getAll({ populate: false }, (windows) => {
-        if (windows.length === 1) {
-            if(data.loggedIn && data.sessionTimeout != true)
-            {restartTimer();}
-            // console.log(data.loggedIn, data.sessionTimeout, data.timeLeft)
-            console.log('First window opened!');
+    }
+    if (data.loggedIn && data.sessionTimeout) {
+        chrome.tabs.update(tab.id, { url: '../content/ui/sessionTimeout.html' });
+    }
+    else if (data.loggedIn && !data.sessionTimeout) {
+        const tabs = await chrome.tabs.query({})
+        if (tabs.length === 1) {
+            startTimer();
         }
-    });
+    }
 });
 
-chrome.tabs.onCreated.addListener((tab) => {
-    chrome.storage.local.get(['loggedIn', 'sessionTimeout'], (data) => {
-        if (data.loggedIn && data.sessionTimeout) {
-            const tabUrl = tab.url;
-            if (tabUrl !== "chrome://extensions/") {
-
-                chrome.tabs.update(tab.id, { url: '../content/ui/sessionTimeout.html' });
-            }
-            blockHttpsSearch();
+// Function to update time in local storage every minute
+let intervalId;
+const updateTimeInLocalStorage = async (timeLeft) => {
+    intervalId = setInterval(async () => {
+        timeLeft -= 20000;
```

Comment:
Avoid using hardcoding and use const variables defined on top of the file for ex.
timeLeft -= BROWSER_TIME_LEFT_LOCAL_STORAGE_UPDATE_FREQUENCY
this is just an example variable names but you can use similar names.